src/Pages/SignUp/SignUp.js

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from "react";
+import StatusView from "./StatusView";
+import Agreement from "./Agreement";
+import "./SignUp.scss";
+
+class SignUp extends Component {
+  constructor() {
+    super();
+    this.state = {
+      allCheckd: false,
+      agreement: [],
+    };
+  }
+  componentDidMount() {
+    let agreement = [];
+    agreementList.forEach((el) => {
+      let obj = {
+        isCheck: false,
+        content: el,
+      };
+      agreement.push(obj);
+    });
+    this.setState({ agreement });
+  }
+
+  isCheck = (e) => {
+    const target = e.target;
+    let agreement = this.state.agreement;
+    let allCheckd = true;
+
+    if (target.name === "all") {
+      allCheckd = !this.state.allCheckd;
+      agreement.map((el) => (el.isCheck = target.checked));
+    } else {
+      agreement.forEach((el, index) => {
+        el.isCheck = index === parseInt(target.name) ? !el.isCheck : el.isCheck;
+        allCheckd = allCheckd && el.isCheck ? true : false;
+      });
+    }
+    this.setState({ agreement, allCheckd });
+  };
+
+  nextSignUp = (e) => {
+    e.preventDefault();
+    const { agreement } = this.state;
+    let comparison = true;
+    agreement.some((el) => {
+      comparison = comparison && el.isCheck ? true : false;
+      if (!comparison) {
+        alert(el.content + "를 확인 해주세요");
+        return true;
+      }
+    });
+    if (comparison) this.props.history.push("/SignUpInfo");
+  };
```

Comment:
로직이 불필요하게 들어있는 것 같습니다!
어떤 로직을 구현하려고 하셨나요!?

모두 체크가 됐을 때 넘어가고 아니면 체크하지 않은 부분의 제일 첫번째를 alert로 띄워주고 싶으시면 다음과 같이 해보세요!
```suggestion

  nextSignUp = e => {
    e.preventDefault();
    const { agreement } = this.state;
    const filtered = agreement.filter(el => !el.isCheck);
    if (filtered.length > 0) {
      alert(`${filtered[0].content} 입력해주세요`);
    } else {
      this.props.history.push("/SignUpInfo");
    }
  };
```

Reply:
역시 알고리즘을 알아야하내요